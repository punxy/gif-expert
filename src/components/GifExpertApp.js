import React, { useState } from 'react'
import { AddCategory } from './AddCategory';
import { GifGrid } from './GifGrid';

export const GifExpertApp = ({ defaultCategories = [] }) => {

    const [categories, setCategories] = useState(defaultCategories)
   
    // const handleAdd = () => {
    //     //setCategories([ ...categories, 'HunterXHunter' ]);
    //     setCategories( cats =>  [...cats, 'HunterXHunter']);
    // }

    return (
        <>
            <h2>GifExpertApp</h2>
            <hr />

            <AddCategory setCategories={setCategories} />

            <ol>
                { categories.map( category => 
                        <GifGrid key={category} category={category} />
                )}
            </ol>
        </>
    )
}
