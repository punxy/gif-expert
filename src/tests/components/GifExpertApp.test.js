import React from 'react';
import { shallow } from "enzyme";
import '@testing-library/jest-dom';

import { GifExpertApp } from '../../components/GifExpertApp';

describe('Test in GifExpertApp component', () => {

    test('should be render correctly', () => {
        const wrapper = shallow( <GifExpertApp  /> );
        expect(wrapper).toMatchSnapshot();
    });

    test('should be display a list of categories', () => {

        const categories = ['OnePunch', 'Drgaon Ball'];
        const wrapper = shallow( <GifExpertApp defaultCategories={categories} /> );
        
        expect(wrapper).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe(categories.length);
    });
});