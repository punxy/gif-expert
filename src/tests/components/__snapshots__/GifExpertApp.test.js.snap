// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in GifExpertApp component should be display a list of categories 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ol>
    <GifGrid
      category="OnePunch"
      key="OnePunch"
    />
    <GifGrid
      category="Drgaon Ball"
      key="Drgaon Ball"
    />
  </ol>
</Fragment>
`;

exports[`Test in GifExpertApp component should be render correctly 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ol />
</Fragment>
`;
